src/brod_utils.erl

```diff
@@ -141,8 +143,17 @@ get_metadata(Hosts, Topics, ConnCfg) ->
                      offset_time(), conn_config()) ->
         {ok, offset()} | {error, any()}.
 resolve_offset(Hosts, Topic, Partition, Time, ConnCfg) ->
+  Opts = #{timeout => proplists:get_value(
+                                  connect_timeout, ConnCfg, ?DEFAULT_TIMEOUT)},
```

Comment:
elvis is not happy about this line being too long (see travis build),
maybe add a variable for the fallback timeout value.